src/sid/update_states.py

```diff
@@ -158,3 +164,16 @@ def _update_info_on_new_tests(
     states.loc[states["received_test_result"], "received_test_result"] = False
 
     return states
+
+
+def _update_info_on_new_vaccinations(
+    states: pd.DataFrame, newly_vaccinated: pd.Series
+) -> pd.DataFrame:
+    """Activate the counter for immunity by vaccinations."""
+    states["newly_vaccinated"] = newly_vaccinated
+    states.loc[newly_vaccinated, "ever_vaccinated"] = newly_vaccinated
```

Comment:
Would `states.loc[newly_vaccinated, "ever_vaccinated"] = True` also work? If so, I find it more readable. I am not even sure why your version runs. If `newly_vaccinated` has False entries, I would expect a length mismatch.

Reply:
I think pandas would match via indices, but I prefer your syntax as well.

Reply:
Actually, it is not a guess. It is tested that it works like this. And its something I appreciated when using other languages.

Reply:
Makes sense. Thanks!